subprojects/workers/src/main/java/org/gradle/workers/internal/ActionExecutionSpec.java

```diff
@@ -16,9 +16,16 @@
 
 package org.gradle.workers.internal;
 
+import org.gradle.internal.reflect.Instantiator;
+
+import javax.annotation.Nullable;
+
 public interface ActionExecutionSpec extends WorkSpec {
     Class<? extends Runnable> getImplementationClass();
 
+    @Nullable
+    Instantiator getActionInstantiator();
```

Comment:
Alternatively we could stop serializing the whole spec. Let's follow up when we actually need classloader isolation.